app/iiif/views.py

```diff
@@ -1,15 +1,37 @@
+import requests
 from django.conf import settings
 from django.http import HttpResponse
 from django.views.decorators.csrf import csrf_exempt
+from . import tools
 
 
 @csrf_exempt
 def index(request, iiif_url):
-    if request.is_authorized_for(settings.EDEPOT_PRIVATE_SCOPE):
-        return HttpResponse("APPROVED")
-    elif request.is_authorized_for(settings.EDEPOT_PUBLIC_SCOPE):
-        # TODO: Get dossiernumber from iiif_url
-        # TODO: Get metadata from stadsarchiefserver
-        return HttpResponse("APPROVED IF IMAGE IS PUBLIC")
-    else:
-        return HttpResponse("DENIED", status=401)
+    try:
+        stadsdeel, dossier, document_barcode, file = tools.get_info_from_iiif_url(iiif_url)
+    except tools.InvalidIIIFUrlError:
+        return HttpResponse("Invalid formatted url", status=400)
+
+    # Get image meta data
+    meta_response = tools.get_meta_data(dossier)
+    if meta_response.status_code != 200:
+        return HttpResponse("No metadata could be found for this image", status=404)
+    # TODO: interpret the xml to get the image "status"
+    metadata = meta_response.json()
+
+    # Get the image itself
+    img_response = tools.get_image_from_iiif_server(stadsdeel, dossier, document_barcode, file)
+    if img_response.status_code != 200:
+        return HttpResponse("No image could be found", status=404)
```

Comment:
I think it's better to return the IIIF server response unchanged because the issue could be with modifying the image. Otherwise it will be really hard to point out where the error is.